### PR TITLE
#5000 Improve fetching remote patient UX

### LINKED ISSUE #5000: Clicking Ok & Next after fetching patient from central goes to new patient creation page
## What went wrong? 😲

https://github.com/user-attachments/assets/e46664ec-3486-4cd0-81cf-057a23022e95

## Expected behaviour 🤔
Should redirect me to the patient that I fetched. (There is a workaround by clicking the downloaded patient in the row on search page does redirect you to correct patient...)

## How to Reproduce 🔨

Steps to reproduce the behaviour:

1. Have patients belonging to a different store
2. Create new patient that has the name of an existing patient from another store
3. Download patient from search step
4. Click Ok & Next
5. See new patient creation page

## Your environment 🌱

- Open mSupply Version: 2.3.0-RC5
- Legacy mSupply Central Server Version:

- Platform:

- Database type:

### PR DESCRIPTION
**Pull request by CarlosNZ**
Fixes #5000 

# 👩🏻‍💻 What does this PR do?

Implements changes as explained [here](https://github.com/msupply-foundation/open-msupply/issues/5000#issuecomment-2461224938)

- Improvements to the button and message text to make it clearer what's going on
- When downloading remote patient, go to patient page after download confirmation rather than leave option to "Create new"

Screenshot 2024-11-08 at 11 56 01 AM

# 🧪 Testing

You'll need to create some patients on mSupply central (4D) belonging to a different store than the currently logged in one.

- [ ] Create "New Patient". On the "search results" steps, button should show "Create new" (which loads new patient input with the search details entered)
- [ ] New Patient => Search for patient that already exists locally. Results should show up in "Search results", and clicking on one takes you to that patient details page
- [ ] New Patient => Search for a patient that only exists on central server. Search results should show "Download" icon next to patient.
  - [ ] Click to download. Confirm fetch. Once downloaded, clicking "View patient" should take you to that patient's detail page. You shouldn't be able to create a new patient at this point

### CODE REVIEW

File: client/packages/common/src/intl/locales/en/common.json [hunk 1]
```diff
@@ -1409,7 +1410,7 @@
   "messages.outbound-shipment-created-on": "Outbound shipment created on {{date}}",
   "messages.over-allocated": "Due to the pack sizes available a total quantity of {{quantity}} has been allocated rather than {{issueQuantity}}",
   "messages.patient-data-required-for-search": "Please enter patient information",
-  "messages.patients-create": "Click [OK & Next] to add a new patient with the details entered, or click an existing patient below to view their details.",
+  "messages.patients-create": "Click [Create new] to add a new patient with the details you entered on the previous tab, or click an existing patient below to view their details.",
```
Comment (CarlosNZ) [inline client/packages/common/src/intl/locales/en/common.json:1413]: I think this is just a little bit clearer

File: client/packages/system/src/Patient/CreatePatientModal/FetchPatientModal.tsx [hunk 2]
```diff
@@ -63,7 +67,8 @@ export const FetchPatientModal: FC<FetchPatientModal> = ({
       title={t('title.patient-retrieval-modal')}
       okButton={
         <DialogButton
-          variant={step === 'Synced' ? 'ok' : 'next'}
+          variant="ok"
+          customLabel={step === 'Synced' ? 'View patient' : undefined}
```
Comment (lache-melvin) [inline client/packages/system/src/Patient/CreatePatientModal/FetchPatientModal.tsx:71]: Should translation the `View patient` here

**Review by lache-melvin — APPROVED** [review]
Soo much nicer Carl!! Thank you - merge when ready, just the one string to translate 🙏

File: client/packages/system/src/Patient/CreatePatientModal/FetchPatientModal.tsx [hunk 3]
```diff
@@ -18,6 +19,7 @@ import {
   Typography,
 } from '@common/components';
 import { useFetchPatient } from './useFetchPatient';
+import { usePatientStore } from 'packages/programs/src';
```
Comment (lache-melvin) [inline client/packages/system/src/Patient/CreatePatientModal/FetchPatientModal.tsx:22]: Ah and also fix this, needs to be `from '@openmsupply-client/programs'` for tests to run!

**Pull request merged**